### PR TITLE
fix: allow to cancel creation of new kubernetes connection

### PR DESCRIPTION
**Pull request by lstocchi**
### What does this PR do?

This PR allows to cancel the creation of a new kind connection. It also fixes a misbehavior on the task manager (if the creation was cancelled the task appeared to be on running)

### Screenshot/screencast of this PR

cancel-kube-creation

### What issues does this PR fix or reference?

it is part of #1337 

### How to test this PR?

1. create a new kind cluster and cancel the creation

### CODE REVIEW

File: extensions/kind/src/util.ts [hunk 1]
```diff
@@ -106,6 +112,18 @@ export function runCliCommand(command: string, args: string[], options?: RunOpti
     }
 
     const spawnProcess = spawn(command, args, { shell: isWindows(), env });
+
+    // if the token is cancelled, kill the process and reject the promise
+    token?.onCancellationRequested(() => {
+      if (isWindows()) {
```
Comment (jeffmaury) [inline extensions/kind/src/util.ts]: For future reuse should be in a utility method

File: packages/renderer/src/lib/preferences/PreferencesConnectionCreationRendering.svelte [hunk 2]
```diff
@@ -164,15 +166,14 @@ function getLoggerHandler(): ConnectionCallback {
 async function ended() {
   creationInProgress = false;
   tokenId = undefined;
-  if (creationCancelled) {
-    // the creation has been cancelled
+  if (creationCancelled || creationFailed) {
```
Comment (jeffmaury) [inline packages/renderer/src/lib/preferences/PreferencesConnectionCreationRendering.svelte]: Can be rewritten as:
```
if (!creationCancelled && !creationFailed) {
    window.dispatchEvent(new CustomEvent('provider-lifecycle-change'));
    creationSuccessful = true;
}
updateStore();
```

**Review by jeffmaury — APPROVED** [review]
LGTM

**Comment by lstocchi** [comment]
@benoitf or @cdrage could you give it a try on mac? On windows the classic `process.kill` didn't work so i had to invoke the taskkill process. If you can confirm the normal way works on mac it would be great. Thanks!

**Review by benoitf — COMMENTED** [review]
https://user-images.githubusercontent.com/436777/230617986-e5ac5f8f-df5b-4c34-9225-f37f59eb10a1.mp4

**Comment by lstocchi** [comment]
> 

Thanks. I'm going to merge it tomorrow if nobody has any issue with it

**Comment by lstocchi** [comment]
@benoitf Sonar gives an error that prevent this PR to be merged but it's a false positive from my pov. 
Basically on windows the shell arg is true so the process returned by spawn is not the actual command to be killed. Taskkill kill it and all its children (which include the actual command process).
As an alternative we could find all children from the process spawned and kill them one by one but is it worth?

**Comment by benoitf** [comment]
@lstocchi you can login into sonarcloud and flag it as false positive

**Comment by lstocchi** [comment]
@benoitf do i need to be enabled somehow to resolve an issue as false positive? I tried logging in using my gh account and i cannot see any button to flag the issue 🤔

**Comment by benoitf** [comment]
I've done the review on SonarCloud
Merging

**Pull request merged**